rb/lib/selenium/webdriver/remote/bridge.rb

```diff
@@ -568,6 +569,52 @@ def shadow_root(element)
           ShadowRoot.new self, shadow_root_id_from(id)
         end
 
+        #
+        # virtual-authenticator
+        #
+
+        def virtual_authenticator_id
+          @authenticator_id
+        end
+
+        def add_virtual_authenticator(options)
+          @authenticator_id = execute :add_virtual_authenticator, {}, options.to_dict
+        end
+
+        def remove_virtual_authenticator
+          execute :remove_virtual_authenticator, {}, {authenticatorId: @authenticator_id}
+          @authenticator_id = nil
+        end
+
+        def add_credential(credential)
+          credential = credential.to_dict
+          credential[:authenticatorId] = @authenticator_id
+          execute :add_credential, {}, credential
+        end
+
+        def credentials
+          credential_data = execute :get_credentials, {}, {authenticatorId: @authenticator_id}
+          credential_list = []
+
+          credential_data.each do |cred|
+            credential_list.push(Credential.new.from_dict(cred))
+          end
+          credential_list
+        end
+
+        def remove_credential(credential_id)
+          credential_id = Base64.urlsafe_encode64(credential_id.pack('C*')) if credential_id.instance_of?(Array)
```

Comment:
How can `credential_id` be an array?

Reply:
`credential_id` can be `id` or `rawID`. For example, this is a credential registered with authenticator.

`{"credential"=>{"id"=>"-lD3cYWt2iGgWuYkEU2FI7GH3h96ULdewKtqAHWL6g8", "rawId"=>[250, 80, 247, 113, 133, 173, 218, 33, 160, 90, 230, 36, 17, 77, 133, 35, 177, 135, 222, 31, 122, 80, 183, 94, 192, 171, 106, 0, 117, 139, 234, 15],...} `

Thus, both `id` and `rawId` can be used to remove a credential. For `rawId` we need to convert it to its matching `id`.

Similar thing has been done in java too. 
https://github.com/SeleniumHQ/selenium/blob/trunk/java/src/org/openqa/selenium/remote/RemoteWebDriver.java#L1218

Reply:
I see, thank you for explanation.